File: python/122. Best Time to Buy and Sell Stock II.py
'''
Say you have an array for which the ith element is the price of a given stock on day i.

Design an algorithm to find the maximum profit. 
You may complete as many transactions as you like (ie, buy one and sell one share of the stock multiple times). 
However, you may not engage in multiple transactions at the same time (ie, you must sell the stock before you buy again).
'''
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        if len(prices) < 2:
            return 0 
        
        max_profit = 0
        for i in range(len(prices)-1):
            price = prices[i+1] - prices[i]
            if price > 0:
                max_profit += price
                
        return max_profit

File: Java/122. Best Time to Buy and Sell Stock II.java
'''
Say you have an array for which the ith element is the price of a given stock on day i.

Design an algorithm to find the maximum profit. 
You may complete as many transactions as you like (ie, buy one and sell one share of the stock multiple times). 
However, you may not engage in multiple transactions at the same time (ie, you must sell the stock before you buy again).
'''

class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length < 2) {
            return 0;
        }
        
        int max_profit = 0;
        
        for (int i = 0; i < (prices.length-1); i++) {
            int price = prices[i+1] - prices[i];
            if (price > 0) {
                max_profit+=price;
            }
        }
        return max_profit;
    }
}
